core/src/main/java/com/sequenceiq/cloudbreak/service/stack/TargetedUpscaleSupportService.java

```diff
@@ -36,29 +36,42 @@
     @Inject
     private StackUtil stackUtil;
 
-    @Cacheable(cacheNames = "targetedUpscaleCache", key = "{ #stack.resourceCrn }")
+    public TargetedUpscaleSupportService() {
+    }
+
     public boolean targetedUpscaleOperationSupported(Stack stack) {
         try {
-            String accountId = Crn.safeFromString(stack.getResourceCrn()).getAccountId();
-            return entitlementService.targetedUpscaleSupported(accountId) &&
-                    isUnboundEliminationSupported(accountId) && isUnboundClusterConfigRemoved(stack);
+            return targetedUpscaleEntitlementsEnabled(stack.getResourceCrn()) && DnsResolverType.FREEIPA.equals(stack.getDomainDnsResolver());
         } catch (Exception e) {
             LOGGER.error("Error occurred during checking if targeted upscale supported, thus assuming it is not enabled, cause: ", e);
             return false;
         }
     }
 
-    private boolean isUnboundClusterConfigRemoved(Stack stack) {
+    public boolean targetedUpscaleEntitlementsEnabled(String crn) {
+        String accountId = Crn.safeFromString(crn).getAccountId();
+        return entitlementService.targetedUpscaleSupported(accountId) && isUnboundEliminationSupported(accountId);
+    }
+
+    public Stack updateDnsResolverType(Stack stack) {
         GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stack);
         Set<Node> reachableNodes = stackUtil.collectReachableNodes(stack);
         Set<String> reachableHostnames = reachableNodes.stream().map(Node::getHostname).collect(Collectors.toSet());
         boolean unboundClusterConfigPresentOnAnyNodes = hostOrchestrator.unboundClusterConfigPresentOnAnyNodes(primaryGatewayConfig, reachableHostnames);
         LOGGER.info("Result of check whether unbound config is present on nodes of stack [{}] is: {}",
                 stack.getResourceCrn(), unboundClusterConfigPresentOnAnyNodes);
-        return !unboundClusterConfigPresentOnAnyNodes;
+        if (unboundClusterConfigPresentOnAnyNodes) {
+            stack.setDomainDnsResolver(DnsResolverType.LOCAL_UNBOUND);
+        } else if (isUnboundEliminationSupported(stack.getResourceCrn())) {
+            stack.setDomainDnsResolver(DnsResolverType.FREEIPA);
+        } else {
+            // if 00-cluster.conf is not present on any node, but unbound elimination is not enabled, then something strange is going on
+            stack.setDomainDnsResolver(DnsResolverType.UNKNOWN);
```

Comment:
could you add a log here to raise attention and then you could remove the comment